exporter/trace/cloudtrace.go

```diff
@@ -267,34 +171,47 @@ type Exporter struct {
 
 // InstallNewPipeline instantiates a NewExportPipeline and registers it globally.
 func InstallNewPipeline(opts []Option, topts ...sdktrace.TracerProviderOption) (trace.TracerProvider, func(), error) {
-	tp, flush, err := NewExportPipeline(opts, topts...)
+	tp, shutdown, err := NewExportPipeline(opts, topts...)
 	if err != nil {
 		return nil, nil, err
 	}
 	otel.SetTracerProvider(tp)
-	return tp, flush, err
+	return tp, shutdown, err
 }
 
 // NewExportPipeline sets up a complete export pipeline with the recommended setup
-// for trace provider. Returns provider, flush function, and errors.
+// for trace provider. Returns provider, shutdown function, and errors.
 func NewExportPipeline(opts []Option, topts ...sdktrace.TracerProviderOption) (trace.TracerProvider, func(), error) {
-	exporter, err := NewExporter(opts...)
+	// TODO(suereth): Don't flesh options twice.
+	o := options{Context: context.Background()}
+	for _, opt := range opts {
+		opt(&o)
+	}
+	exporter, err := NewExporterWithOptions(&o)
 	if err != nil {
 		return nil, nil, err
 	}
-	tp := sdktrace.NewTracerProvider(append(topts, sdktrace.WithSyncer(exporter))...)
-	return tp, exporter.traceExporter.Flush, nil
+	tp := sdktrace.NewTracerProvider(
+		append(topts,
+			sdktrace.WithBatcher(exporter, o.BatchSpanProcessorOptions...))...)
+	return tp, func() {
+		tp.Shutdown(context.Background())
+	}, nil
 }
 
 // NewExporter creates a new Exporter thats implements trace.Exporter.
-//
-// TODO(yoshifumi): add a metrics exporter one the spec definition
-// process and the sampler implementation are done.
 func NewExporter(opts ...Option) (*Exporter, error) {
 	o := options{Context: context.Background()}
 	for _, opt := range opts {
 		opt(&o)
 	}
+	return NewExporterWithOptions(&o)
+}
+
+// NewExporter creates a new Exporter thats implements trace.Exporter.
+// This version takes a single options struct.  This shouldn't be called
```

Comment:
If this shouldn't be called by user code, should this function be package-local (private)?

Reply:
Oh my, just learned about capitals vs. not capitals.... will fix.